task/github-sarif-upload/0.1/github-sarif-upload.yaml

```diff
@@ -0,0 +1,277 @@
+        echo ""
+
+        # Check if the output contains any manifests
+        if echo "${discover_output}" | grep -q '"manifests"'; then
+          echo "[$(date --utc -Ins)] Found manifests in output"
+        else
+          echo "[$(date --utc -Ins)] No manifests found in output"
+          echo "[$(date --utc -Ins)] No SARIF artifacts found in image ${IMAGE_URL}@${IMAGE_DIGEST}"
+          echo "NO_ARTIFACTS" > /var/workdir/artifact_status.txt
+          exit 0
+        fi
+
+        # Extract digests using grep and cut to handle multiline JSON
+        echo "[$(date --utc -Ins)] Extracting digests from oras output..."
+        digests=$(echo "${discover_output}" | grep '"digest"' | cut -d'"' -f4)
+
+        if [ -z "${digests}" ]; then
+          echo "[$(date --utc -Ins)] No digests found in manifests"
+          echo "NO_ARTIFACTS" > /var/workdir/artifact_status.txt
+          exit 0
+        fi
+
+        echo "[$(date --utc -Ins)] Found digests:"
+        echo "${digests}"
+
+        # Create directory for the artifacts
+        mkdir -p /var/workdir/sarif-artifacts
+
+        # Use oras to pull artifacts
+        echo "[$(date --utc -Ins)] Pulling SARIF artifacts using oras..."
+
+        for digest in ${digests}; do
+          echo "[$(date --utc -Ins)] Pulling artifact with digest: ${digest}"
+
+          # Use oras to pull the artifact
+          if oras pull --no-tty --registry-config "${HOME}/auth.json" "${IMAGE_URL}@${digest}" --output /var/workdir/sarif-artifacts/; then
+            echo "[$(date --utc -Ins)] Successfully pulled artifact ${digest}"
+          else
+            echo "ERROR: Failed to pull artifact ${digest}" >&2
+            exit 1
+          fi
+        done
+
+        # Find all SARIF files in the pulled artifacts
+        echo "[$(date --utc -Ins)] Looking for SARIF files in /var/workdir/sarif-artifacts..."
+        ls -la /var/workdir/sarif-artifacts/
+
+        sarif_files=""
+        shopt -s nullglob
+        for file in /var/workdir/sarif-artifacts/*.sarif; do
+          sarif_files="${sarif_files} $(basename "${file}")"
+        done
+        sarif_files="${sarif_files# }"
+        shopt -u nullglob
+
+        if [ -n "${sarif_files}" ]; then
+          echo "[$(date --utc -Ins)] Found SARIF files:"
+          echo "${sarif_files}"
+          echo "ARTIFACTS_FOUND" > /var/workdir/artifact_status.txt
+        else
+          echo "ERROR: No SARIF files found in pulled artifacts" >&2
+          exit 1
+        fi
+
+    - name: upload-sarif
+      image: quay.io/konflux-ci/konflux-test:v1.4.40@sha256:99eb8bcc7bcb35bdd5edea7b0ac333bbdb67586dea6b4dab92baf2b8fb32bf2c
+      workingDir: /var/workdir
+      script: |
+        #!/usr/bin/env bash
+
+        set -euo pipefail
+        # shellcheck source=/dev/null
+        . /utils.sh
+        trap 'handle_error $(results.TEST_OUTPUT.path)' EXIT
+
+        echo "[$(date --utc -Ins)] Fix permissions on Java binaries in CodeQL bundle"
```

Comment:
Is there a reason to update the Java binary permissions here instead of when the konflux-test image is built?

Reply:
@nmars no reason. I changed permissions on the codeql binary in konflux-test and released it. Only to realize it also needs executable permissions on the java binary. 

In order to not force another release for a small change, I'll submit a PR to konflux-test and then later remove these lines once the image is released the next time/

Reply:
On second thoughts, from a security standpoint, maybe it makes sense to only change the permissions on the Java binary at the task level. thoughts?